src/fclone.js

```diff
@@ -20,7 +20,7 @@ function fclone(obj, refs) {
   }
 
   // typed array Int32Array etc.
-  if (/\d[a-z]*Array]$/i.test(Object.prototype.toString.call(obj))) {
+  if (obj.subarray  && typeof obj.subarray === 'function' && /^[A-Z][A-Za-z\d]+Array$/.test(Object.prototype.toString.call(obj))) {
```

Comment:
Didn't mean to change the regexp, the subarray check was in order to verify the method before trying to call it, also, likely faster invalidation than the regex check.